vector-search.mdx

```diff
@@ -0,0 +1,156 @@
+---
+title: Nearest Neighbors (Vector Search)
+description: Find the nearest neighbors across a vector relationship
+comingSoon: true
+---
+
+import { TipInfo, TipBad, TipGood } from '@/components/Tip'
+
+---
+
+A feature set can be linked to the closest examples of another feature set. This functionality can be useful for search and retrieval applications.
+
+Nearest neighbor relationships are only supported for [vector features](/docs/feature-types#vectors).
+
+<TipInfo>
+  We recommend to first take a look through Chalk's support for <a href="/docs/feature-types#vectors">vector
+  features</a> and <a href="/docs/embeddings">embedding functions</a>.
+</TipInfo>
+
+
+To illustrate how to use nearest neighbor relationships, we'll walk through an example for Chalk can power FAQ search.
+In this example, the feature set `SearchQuery` feature set represents an incoming request, and the `FAQDocument` feature set represents our collection
+of frequently asked questions and answers. Our goal is to return the five most relevant FAQ entries for the given search query.
+
+---
+
+## Defining nearest neighbor relationships
+Using the `has_many` function and the `isnear` method, we can express a relationship where we want the nearest documents for each query.
+
+```py
+from chalk.features import DataFrame, features, embedding, has_many, Vector
+
+@features
+class SearchQuery:
+  query: str
+  product_version: int
+  embedding: Vector = embedding(...)
+  nearest_documents: DataFrame[FAQDocument] = has_many(lambda: SearchQuery.embedding.isnear(FAQDocument.embedding))
```

Comment:
maybe `is_near`? I think people often migrate from the no-underscore to underscore (see polars `groupby` -> `group_by`)

Reply:
Sounds good; will go with `is_near`